File: app/src/main/java/com/prideven/android/HungryEats/menuItems/MenuViewModel.java
package com.prideven.android.hungryeats.menuitems;

import android.util.Log;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.prideven.android.hungryeats.HungryEatsApplication;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;

public class MenuViewModel extends ViewModel {
    // TODO: Implement the ViewModel

    static final String TAG = MenuViewModel.class.getSimpleName();
    HungryEatsApplication hungryEatsApplication = HungryEatsApplication.getInstance();
    // Create a LiveData with a String
    private MutableLiveData<EatsMenuResponse> eatsMenuResponse;

    public MutableLiveData<EatsMenuResponse> getMenuDetails() {
        if (eatsMenuResponse == null) {
            eatsMenuResponse = new MutableLiveData<EatsMenuResponse>();
        }
        return eatsMenuResponse;
    }

    public void fetchMenuList(Callback<EatsMenuResponse> responseCallback,int restaurant_id) {
        Retrofit retrofit=hungryEatsApplication.getRetrofitInstance();
        MenuWebService eatsMenuWebService = retrofit.create(MenuWebService.class);
        Call<EatsMenuResponse> call = eatsMenuWebService.getMenu(restaurant_id);
        call.enqueue(responseCallback);
    }

    public void callMenuDataRepo(int restaurant_id) {

        Callback<EatsMenuResponse> object = new Callback<EatsMenuResponse>() {
            @Override
            public void onResponse(Call<EatsMenuResponse> call, Response<EatsMenuResponse> response) {
                Log.i("", "sucessful");
                eatsMenuResponse.postValue(response.body());

            }
            @Override
            public void onFailure(Call<EatsMenuResponse> call, Throwable throwable) {
                Log.e(TAG, throwable.toString());
            }
        };
        fetchMenuList(object,restaurant_id);
    }
}

File: app/src/main/java/com/prideven/android/HungryEats/menuItems/MenuViewModel.kt
package com.prideven.android.hungryeats.menuitems

import android.util.Log
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.prideven.android.hungryeats.HungryEatsApplication
import com.prideven.android.hungryeats.menuitems.MenuViewModel
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MenuViewModel : ViewModel() {
    var hungryEatsApplication = HungryEatsApplication.getInstance()

    // Create a LiveData with a String
    private var eatsMenuResponse: MutableLiveData<EatsMenuResponse?>? = null
    val menuDetails: MutableLiveData<EatsMenuResponse?>
        get() {
            if (eatsMenuResponse == null) {
                eatsMenuResponse = MutableLiveData()
            }
            return eatsMenuResponse!!
        }

    fun fetchMenuList(
        responseCallback: Callback<EatsMenuResponse>?,
        restaurant_id: Int
    ) {
        val retrofit = hungryEatsApplication.retrofitInstance
        val eatsMenuWebService: MenuWebService = retrofit.create(MenuWebService::class.java)
        val call: Call<EatsMenuResponse> = eatsMenuWebService.getMenu(restaurant_id)
        call.enqueue(responseCallback)
    }

    fun callMenuDataRepo(restaurant_id: Int) {
        val `object`: Callback<EatsMenuResponse> =
            object : Callback<EatsMenuResponse?> {
                override fun onResponse(
                    call: Call<EatsMenuResponse?>,
                    response: Response<EatsMenuResponse?>
                ) {
                    Log.i("", "sucessful")
                    eatsMenuResponse!!.postValue(response.body())
                }

                override fun onFailure(
                    call: Call<EatsMenuResponse?>,
                    throwable: Throwable
                ) {
                    Log.e(TAG, throwable.toString())
                }
            }
        fetchMenuList(`object`, restaurant_id)
    }

    companion object {
        // TODO: Implement the ViewModel
        val TAG = MenuViewModel::class.java.simpleName
    }
}